scripts/datasets/DET.py

```diff
@@ -0,0 +1,203 @@
+"""Prepare DET(Object detection) datasets"""
+import os
+import argparse
+import tarfile
+from os.path import join, isdir
+import glob
+import xml.etree.ElementTree as ET
+import json
+import time
+import sys
+from gluoncv.utils.filesystem import try_import_cv2
+from concurrent import futures
+from gluoncv.utils import download, makedirs
+cv2 = try_import_cv2()
+def parse_args():
+    """dataset parameter."""
+    parser = argparse.ArgumentParser(
+        description='Download DET dataset and prepare for tracking')
+    parser.add_argument('--download-dir', type=str, default='~/.mxnet/datasets/DET/',
+                        help='dataset directory on disk')
+    parser.add_argument('--instanc-size', type=int, default = 511, help='instanc image size')
+    parser.add_argument('--num-threads', type=int, default = 12, help='threads number')
+    args = parser.parse_args()
+    args.download_dir = os.path.expanduser(args.download_dir)
+    return args
+
+def download_det(args,overwrite=False):
+    url = 'http://image-net.org/image/ILSVRC2015/ILSVRC2015_DET.tar.gz'
+    makedirs(args.download_dir)
+    filename = download(url, path=args.download_dir, overwrite=overwrite)
+    with tarfile.open(filename) as tar:
+        tar.extractall(path=args.download_dir)
+    os.rename(join(args.download_dir,'ILSVRC2015'),join(args.download_dir,'ILSVRC'))
+    
+
+def printProgress(iteration, total, prefix='', suffix='', decimals=1, barLength=100):
+    """
+    Call in a loop to create terminal progress bar
+
+    function
+    ----------
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        barLength   - Optional  : character length of bar (Int)
+    """
+    formatStr       = "{0:." + str(decimals) + "f}"
+    percents        = formatStr.format(100 * (iteration / float(total)))
+    filledLength    = int(round(barLength * iteration / float(total)))
+    bar             = '' * filledLength + '-' * (barLength - filledLength)
+    sys.stdout.write('\r%s |%s| %s%s %s' % (prefix, bar, percents, '%', suffix)),
+    if iteration == total:
+        sys.stdout.write('\x1b[2K\r')
+    sys.stdout.flush()
+
+def crop_hwc(image, bbox, out_sz, padding=(0, 0, 0)):
```

Comment:
Add method docstring, what is the input, what is the output, what is the functionality of this method. Provide method docstring for all methods below.